server/consumer.go

```diff
@@ -399,6 +399,22 @@ func setConsumerConfigDefaults(config *ConsumerConfig, lim *JSLimitOpts, accLim
 	if config.DeliverSubject == _EMPTY_ && config.MaxRequestBatch == 0 && lim.MaxRequestBatch > 0 {
 		config.MaxRequestBatch = lim.MaxRequestBatch
 	}
+	if config.MaxAckPending == 0 {
+		config.MaxAckPending = streamCfg.LimitMaxAckPending
+	}
+	if config.InactiveThreshold == 0 {
+		config.InactiveThreshold = streamCfg.LimitInactiveThreshold
+	}
```

Comment:
I am very weary of this kind of thing for a number of reasons:

- Code that used to create a consumer with X properties now created one with Y properties leading to invalidation of testing efforts in hard to discover ways and potential change of behavior by simply upgrading a server
- Systems like terraform that are desired state based will set InactiveThreshold=0 but then when they load the consumer and compare suddenly its the limit from the stream.  It wil then enter a remediation phase setting it to 0, which again gets changed and so forth forever.  And if the property is not an editable property it will destroy and recreate the consumer each time
- The usual action-at-a-distance and unintended consequences problems

Reply:
There are other defaults like max waiting and ack wait, what do we do in those cases?

Reply:
Arent those static defaults thats always the same? So easily handlable in all clients by using same defaults - if I am thinking of the right ones?  Here we have adjustable defaults that can change at any time and are user supplied.